meson.build

```diff
@@ -0,0 +1,1449 @@
+#                 information in the build in "release" as we would in "development/debug", on the grounds that it can
+#                 only help if an end user hits a core-dumping bug.
+#                    Meson encourages you to use either the buildtype option or the debug and optimization options
+#                 rather than setting compiler options directly.  However, this does not give us as much control as we
+#                 would like over compiler flags.  Eg switching 'debug' to 'true' turns on the '-g' flag (equivalent to
+#                 '-g2') on GCC, but there isn't a way via the meson  options to set '-g3' for GCC.  So, we set
+#                 'buildtype=plain' and manage compiler flags directly.
+#
+#
+project('brewtarget', 'cpp',
+        version: '3.0.7',
+        license: 'GPL-3.0-or-later',
+        meson_version: '>=0.60.0',
+        default_options : ['cpp_std=c++20',
+                           'warning_level=3',
+#                           'prefer_static=true',  See comment above for why this is commented-out for now
+                           'buildtype=plain'])
+
+#
+# Although Meson itself is written in Python, Meson build files uses a slihgtly different syntax and have less
+# functionality than Python.  See
+# https://mesonbuild.com/FAQ.html#why-is-meson-not-just-a-python-module-so-i-could-code-my-build-setup-in-python and
+# links therefrom for the rationale for avoiding being a full programming language.
+#
+# Besides some (sometimes annoying) variations in syntax, this also means that you sometimes have to do things in a
+# slightly more cumbersome way than you would in a Python script.  Eg here, in regular Python, we would write:
+#    capitalisedProjectName = projectName.capitalize()
+# But meson.project_name() returns a Meson String, not a Python String, so there's a bit more work to do to get the same
+# result.
+#
+projectName = meson.project_name().to_lower()
+capitalisedProjectName = projectName.substring(0, 1).to_upper() + projectName.substring(1)
+
+message('⭐ Building', projectName, 'version', meson.project_version(), 'for', host_machine.system(),
+        'on', build_machine.system(), 'in', meson.project_source_root(), '⭐')
+compiler = meson.get_compiler('cpp')
+message('Using Meson', meson.version(), ', ', compiler.get_id(), 'compiler and', compiler.get_linker_id(), 'linker')
+
+#
+# We need two versions of the main executable name because they are different on Windows:
+#    - We need a "target" name _without_ the '.exe' suffix for the executable() command because, on Windows, Meson will
+#      always add its own '.exe' suffix and we don't want to end up with '.exe.exe' as the suffix!
+#    - Everywhere else, we want the actual name of the file, including the '.exe' suffix.
+#
+# The default (on Windows and Linux) is to use Unix-style executable names, ie all lower case
```

Comment:
Do you need the basename or is the full filename okay? You can calculate the target name, then take the returned `executable()` object and use its `.full_path()` method to acquire the string value of the output filename, prefixed by the full builddir path.

Reply:
Good to know about `.full_path()`.  Will update comment.